modules/services/swayosd.nix

```
@@ -26,6 +26,15 @@ in {
       description = "OSD margin from top edge (0.5 would be screen center).";
     };
 
+    style = mkOption {
```

Perhaps more clear to include "path" in the name:

```suggestion
    stylePath = mkOption {
```

---

Alternatively, can leave it as `style` and use the type `types.lines`. The user could then either provide the style inline, or provide it through a file through `builtins.readFile`.

---

> Alternatively, can leave it as `style` and use the type `types.lines`. The user could then either provide the style inline, or provide it through a file through `builtins.readFile`.

I was just thinking about keeping both, but the `style` inline would have to store it in a file and provide that path, I think this would complicate things a bit because it raises the question of `Where do we store the content?`, perhaps where we store it by default won't be appreciated by the users, for now I'll resolve the other suggestions.

---

In the context of Nix, the answer to the "Where do we store the content?" is always "the Nix store" 🙂 With pretty much the only exception being sensitive information such as passwords.

In any case, having `stylePath` is fine. If somebody wants to merge multiple style definitions, then perhaps can consider switching to lines in the future.